### PR TITLE
Use stevedore to load extractors

### LINKED ISSUE #31: Use "stevedore" to load classes
We are using a simple classloader for loading the extractors, we should rather move to stevedore and define entrypoints for the classes, instead of referencing them by the full namespace.

### PR DESCRIPTION
**Pull request by aidaph**
This commit Fixes #31 

- Load extractors by stevedore plugin from Openstack

### CODE REVIEW

File: caso/extract/manager.py [hunk 1]
```diff
@@ -21,11 +21,7 @@
 from oslo_log import log
 from oslo_utils import importutils
 import six
-
-SUPPORTED_EXTRACTORS = {
-    'nova': 'caso.extract.nova.OpenStackExtractor',
-    'ceilometer': 'caso.extract.ceilometer.CeilometerExtractor',
-}
+from stevedore.extension import ExtensionManager
```
Comment (alvarolopez) [inline caso/extract/manager.py]: Add a ```NAMESPACE = "caso.extractor"``` here and use it below

File: caso/extract/manager.py [hunk 2]
```diff
@@ -21,11 +21,7 @@
...
```
Comment (alvarolopez) [inline caso/extract/manager.py]: Do not import classes but modules:

```
from stevedore import extension
```

File: setup.cfg [hunk 3]
```diff
@@ -54,3 +54,8 @@ input_file = caso/locale/caso.pot
 keywords = _ gettext ngettext l_ lazy_gettext
 mapping_file = babel.cfg
 output_file = caso/locale/caso.pot
+
+[entry_points]
```
Comment (alvarolopez) [inline setup.cfg]: Move this to the entry points above

File: caso/extract/manager.py [hunk 4]
```diff
@@ -19,13 +19,12 @@
 from dateutil import tz
 from oslo_config import cfg
 from oslo_log import log
-from oslo_utils import importutils
 import six
+from stevedore import extension
+
+manager = extension.ExtensionManager(namespace='extractors',
```
Comment (alvarolopez) [inline caso/extract/manager.py]: I will put the namespace in a global NAMESPACE variable, and use "caso.extractors" as its value.

File: setup.cfg [hunk 5]
```diff
@@ -22,17 +22,21 @@ classifier =
...
 [entry_points]
 
 oslo.config.opts =
     caso = caso.opts:list_opts
 
-console_scripts = 
+console_scripts =
     caso-extract = caso.cmd.extract:main
 
+extractors =
```
Comment (alvarolopez) [inline setup.cfg]: rename to caso.extractors

**Comment by dougszumski** [comment]
Is anyone still working on this?

**Comment by alvarolopez** [comment]
@dougszumski I think that @aidaph was working, but other higher priority issues have arrived.

**Pull request closed without merging**